src/backend/vulkan/src/physical_device.rs

```diff
@@ -734,17 +734,36 @@ pub(crate) fn load_adapter(
     };
 
     let queue_families = unsafe {
-        instance
+        let queue_families = instance
             .inner
             .get_physical_device_queue_family_properties(device)
             .into_iter()
-            .enumerate()
-            .map(|(i, properties)| QueueFamily {
-                properties,
-                device,
-                index: i as u32,
-            })
-            .collect()
+            .enumerate();
+
+        if xr::in_use() {
+            queue_families
+                .filter(|(_, info)| {
+                    if info.queue_flags.contains(vk::QueueFlags::GRAPHICS) {
```

Comment:
similarly, this should be a decision of the user of gfx-hal, not the library itself

Reply:
Would that mean e.g. wgpu?

Reply:
correct, that would be `wgpu`